waveorder/io/reader_base.py

```diff
@@ -1,8 +1,25 @@
 import zarr
 import numpy as np
 
+
 class ReaderBase:
 
+    def __init__(self):
+        self.frames = None
+        self.channels = None
+        self.slices = None
+        self.height = None
+        self.width = None
+        self.dtype = None
+        self.mm_meta = None
+        self.stage_positions = None
+        self.z_step_size = None
+        self.channel_names = None
```

Comment:
I'm less clear about the motivation for these lines since they're not used in this parent class? Is is just so that you can be sure that all of the readers have something in these fields?

Reply:
I don't fully remember, but I think this means that we expect all readers which inherit this base class to have these attributes. We check for these attributes in our tests